ocaml/xapi/xapi_vdi.ml

```diff
@@ -905,31 +910,24 @@ let wait_for_vbds_to_be_unplugged_and_destroyed ~__context ~self ~timeout =
     (from.Event_types.token, most_recent_vbds_field from.Event_types.events)
   in
   (* Wait for 4 seconds in total for all the VBDs of this VDI to be unplugged & destroyed *)
-  let start = Mtime_clock.now () in
-  let finish =
-    let maybe_finish =
-      let timeout = Mtime.Span.of_uint64_ns (Int64.of_float (timeout *. 1e9)) in
-      Mtime.(add_span start timeout)
-    in
-    (* It is safe to unbox this because the timeout should not cause an overflow *)
-    Option.get maybe_finish
-  in
+  let from_start = Mtime_clock.counter () in
   let token, initial_vbds = next_token_and_vbds ~token:"" ~timeout in
   (* When we use an empty token, we always get back the whole VDI record *)
   let initial_vbds = Option.get initial_vbds in
+  let pp_time () = Fmt.str "%a" Mtime.Span.pp in
   let rec loop ~token ~remaining_vbds =
-    let now = Mtime_clock.now () in
-    if remaining_vbds <> [] && Mtime.is_earlier now ~than:finish then (
+    let elapsed = Mtime_clock.count from_start in
+    if remaining_vbds <> [] && Mtime.Span.compare elapsed timeout < 0 then (
```

Comment:
I quite liked the 'is_earlier', and it still seems to be present in Mtime 2.0, why do we have to use compare instead?

Reply:
These are timespans, not timestamps.
I agree the comparisons should use be more ergonomic. I think longer and shorter are a very good match (and longest, shortest). I want to work more on the conversion before proposing the change upstream to make the comparisons easier to understand

Reply:
Opened the PR upstream https://github.com/dbuenzli/mtime/pull/49

Reply:
It was accepted. Quick turnaround

Reply:
Indeed, the code here can be changed once we take in the (still unreleased) version that adds these functions